lib/propmangle.js

```diff
@@ -158,31 +158,45 @@ function mangle_properties(ast, options) {
     // step 1: find candidates to mangle
     ast.walk(new TreeWalker(function(node){
         if (node instanceof AST_ObjectKeyVal) {
-            add(node.key);
+            if (options.keep_quoted !== "strict" || !node.quote) {
```

Comment:
This condition and the others can be made more efficient by declaring the following above the step 1 `ast.walk`:
```js
    var keep_quoted_strict = options.keep_quoted === "strict";
```